v4/export/sql_type.go

```diff
@@ -101,6 +102,64 @@ func escape(s []byte, bf *bytes.Buffer, escapeQuotation []byte) {
 	}
 }
 
+func escapeSQL(s []byte, bf *bytes.Buffer, escapeBackslash bool) { // revive:disable-line:flag-parameter
+	if escapeBackslash {
+		escapeBackslashFn(s, bf, nil, nil)
+	} else {
+		bf.Write(bytes.ReplaceAll(s, quotationMark, twoQuotationMarks))
+	}
+}
+
+func escapeCSV(s []byte, bf *bytes.Buffer, escapeBackslash bool, delimiter []byte, separator []byte) { // revive:disable-line:flag-parameter
+	if escapeBackslash {
+		escapeBackslashFn(s, bf, delimiter, separator)
+		return
+	}
+
+	var (
+		escape   bool
+		last          = 0
+		separate byte = 0
+	)
+	if len(delimiter) > 0 {
+		bf.Write(bytes.ReplaceAll(s, delimiter, append(delimiter, delimiter...)))
+		return
+	}
+	if len(separator) > 0 {
+		separate = separator[0]
+	}
+
+	for i := 0; i < len(s); i++ {
```

Comment:
i don't think we should care when `escapeBackslash` is false (when the default is true). if the user explicitly provided `--csv-delimiter="" --escape-backslash=false`, it is *the user's fault* that the data contains a `csv-separator`.

this has the same behavior as `SELECT INTO OUTFILE`:

```sql
values row('foo,bar', 'baz'), row('foo', 'bar,baz') 
    into outfile '1.txt' 
    fields terminated by ',' escaped by '';
```

result:

```
foo,bar,baz
foo,bar,baz
```

Reply:
I see. Do we need to escape all the characters that SQL escaped? Or we just need to escape byte 0 like TiDB? https://github.com/pingcap/tidb/blob/c33e90a7aef495bb2e86311b4c62ee98f30ae864/executor/select_into.go#L98

Reply:
when `escapeBlackslash` is true, we must always escape 0, the line terminators, the delimiter, and the separator (when the delimiter is empty). the rest are just bonus (SELECT INTO does not escape them).

when `escapeBlackslash` is false, nothing should be escaped, not even 0.